test/test_imports.py

```diff
@@ -0,0 +1,64 @@
+import ast
+import os
+from _ast import Import, ImportFrom
+from pathlib import Path
+
+import isort
+import toml
+from pyproject_parser import PyProject
+
+
+def test_for_missing_dependencies() -> None:
+    # All tools out there are too powerful because they find all the imports in the haystack package
+    # We need to find only the top level imports
+
+    #### Collect imports
+    top_level_imports = set()
+    for root, dirs, files in os.walk("haystack"):
+        for file in files:
+            path = Path(root) / file
+
+            if not path.suffix.endswith(".py"):
+                continue
```

Comment:
I think we can simplify this a bit with `pathlib.Path`.

Something like `Path("haystack").glob("**/*.py")` should do the trick. 🤔

Reply:
we already have the files from the tree walk so `glob` in my opinion doesn't get us much but happy to adapt

Reply:
ooh, didn't realize it can do recursion 😍

Reply:
done